src/provisioning/request.ts

```diff
@@ -1,16 +1,51 @@
 import Logging, { LogWrapper } from "../components/logging";
 import crypto from "crypto";
 import { ThinRequest } from "..";
+import { Request } from "express";
+import { ParsedQs } from "qs";
+
+// Methods supported by a express.Router
+export type Methods = 'all' |
+'get' |
+'post' |
+'put' |
+'delete' |
+'patch' |
+'options' |
+'head' |
+'checkout' |
+'connect' |
+'copy' |
+'lock' |
+'merge' |
+'mkactivity' |
+'mkcol' |
+'move' |
+'m-search' |
+'notify' |
+'propfind' |
+'proppatch' |
+'purge' |
+'report' |
+'search' |
+'subscribe' |
+'trace' |
+'unlock' |
+'unsubscribe';
 
 export class ProvisioningRequest<
-    Body = Record<string, unknown>, Params = Record<string, unknown>
-    > implements ThinRequest {
+    // These types are taken from express.Request
+    Params = {[key: string]: string},
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ResBody = any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ReqBody = any,
+    ReqQuery = ParsedQs> implements ThinRequest {
     public readonly log: LogWrapper;
     public readonly id: string;
 
-
     constructor(
-        private expressReq: {body: Body, params: Params, path?: string},
+        public readonly expressReq: Request<Params, ResBody, ReqBody, ReqQuery>,
```

Comment:
Exposing this publicly because I think it's probably going to end up useful eventually.